lib/utils/combineJSON.js

```diff
@@ -35,12 +36,14 @@ function combineJSON(arr, deep, collision) {
   }
 
   for (i = 0; i < files.length; i++) {
-    var file_content = fs.readFileSync(files[i], 'utf8');
+    var resolvedPath = resolveCwd(path.isAbsolute(files[i]) ? files[i] : '.' + path.sep + files[i]);
+    var file_content;
 
     try {
-      file_content = JSON.parse(file_content);
+      file_content = require(resolvedPath);
     } catch (e) {
-      throw new SyntaxError(files[i] + ' - failed to parse JSON: ' + e.message);
+      e.message = 'Failed to load or parse JSON or JS Object: ' + e.message;
+      throw e;
```

Comment:
See my outdated comment https://github.com/amzn/style-dictionary/pull/89#pullrequestreview-121910762 for an explanation. I think JSON error handling worked fine, but for all the kinds of errors that can be made in node modules the extra error detail is helpful.

Reply:
JSON error example with new... file name is in the existing e.message:

```
/Users/e828613/sc/manhattan/mds-tokens/node_modules/style-dictionary/lib/utils/combineJSON.js:46
      throw e;
      ^

SyntaxError: Failed to load or parse JSON or JS Object: /Users/e828613/sc/manhattan/mds-tokens/src/tokens/options/color.json: Unexpected token , in JSON at position 1816
    at JSON.parse (<anonymous>)
    at Object.Module._extensions..json (module.js:671:27)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at combineJSON (/Users/e828613/sc/manhattan/mds-tokens/node_modules/style-dictionary/lib/utils/combineJSON.js:43:22)
    at Object.extend (/Users/e828613/sc/manhattan/mds-tokens/node_modules/style-dictionary/lib/extend.js:113:17)
    at Object.<anonymous> (/Users/e828613/sc/manhattan/mds-tokens/src/dictionary-config/style-dictionary.js:1:117)
```

Reply:
Perfect, thanks!